code/game/area/areas.dm

```diff
@@ -498,11 +498,12 @@ GLOBAL_LIST_EMPTY(teleportlocs)
 /**
   * Called when the area power status changes
   *
-  * Updates the area icon and calls power change on all machinees in the area
+  * Updates the area icon, calls power change on all machinees in the area, and sends the `COMSIG_AREA_POWER_CHANGE` signal.
   */
 /area/proc/power_change()
 	for(var/obj/machinery/M in src)	// for each machine in the area
 		M.power_change()				// reverify power status (to update icons etc.)
+	SEND_SIGNAL(src, COMSIG_AREA_POWER_CHANGE)
```

Comment:
Why now just do the same for `/obj/machinery` as well?

Reply:
I'm not sure what you mean here. Make machines use the signal too?

Reply:
Correct. Instead of having it loop through all machines in `src` and calling `power_change` on them, just have them listen for a signal as well.